### DISCUSSION TITLE
Upload files to MISC folder with code

### DISCUSSION THREAD

**Discussion by Fougl** [gopro/labs#443] [General]
Hi, Is there a way to upload files (especially mask for motion trigger) to MISC folder with code, like opengopro. I will have my camera charging all the time, so uploading through usb is not possible. Thanks

**Comment by dnewman-gpsw** [comment]
I don't know what level of folder access is available through OpenGoPro.  Maybe send a question to https://github.com/gopro/OpenGoPro/discussions

**Reply by Fougl** [reply]
Thanks. Is there a chance to feed the gopro the QR code representation through Raspberry Pi? Or the only way to do it is to scan the code, and be present in person to do it?

**Reply by dnewman-gpsw** [reply]
Only two ways are using a QR Code, or writing to the SD with it out of the camera.

**Comment by The-Skunk** [comment]
> 

You can connect the camera through USB for charging and data if you are using a y-splitter, which has on the one side a USB-PD female and on the other a USB data female port. On the cam side, you have on male plug - like this here:  https://www.amazon.de/dp/B0B7GC7CFR?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Reply by Fougl** [reply]
Amazing. Thanks. Have you tested this personally?

**Reply by Fougl** [reply]
could you elaborate? becuase im not too sure, you know what you are talking about. I am still confused how I would disconnect the data part with the code, so that the gopro goes back into recording mode

**Comment by The-Skunk** [comment]
> 



> could you elaborate? becuase im not too sure, you know what you are talking about. I am still confused how I would disconnect the data part with the code, so that the gopro goes back into recording mode

Yes, I have this in production already and, I'm sure what I'm talking about. We are talking here about a hardware, a USB Y-splitter cable (see previous link of Amazon), right? Did you visit Amazon? (you can also use this for searching "AD ADTRIP USB C Splitter")
 
1)  You connect with this cable the (only) male plug to the camera
2) You connect to the female plug of this USB Y splitter cable, where the "headphone" symbol is, another USB-C cable leading to your computers USB port (where you normally would plug the cable of the camera)
3) You connect to the second female plug of this USB Y splitter cable, where the "lightning symbol" is (sometimes also "PD"), another USB-C cable, leading to some sort of power source which can provide 5 VDC and >=2 A.

**Reply by The-Skunk** [reply]
... and later, go to: https://gopro.github.io/labs/control/extensions/ and enter TUSB and 1 to activate it in the firmware.

But, just a suggestion: We let the batteries inside the camera also when USB power is provided. The cameras could have still some sort of power loss (several times happened) and the batteries will buffer this issue.

**Reply by The-Skunk** [reply]
And finally, because you can charge and have at the same time a data connection, you can upload your files every time, also when USB power is connected ;-)

**Reply by Fougl** [reply]
I need to clarify this before I order the cable. Gopro everytime goes into this mode when it gets connected to pc through usb
![image](https://github.com/gopro/labs/assets/21178929/e8a3751d-0dd9-4b42-8074-0a571b225fe7)

And then you can see the following on the PC. No MISC folder!
![image](https://github.com/gopro/labs/assets/21178929/1b17acfd-51a1-4565-8b35-b62718e66099)

The misc folder is only visible when you use SD card and plug it in directly to the PC. I need to access this MISC folder (to upload the mask for motion trigger, becuase it involves a bit of testing), and at the same time avoid the USB connected mode. You are saying that that USB Y splitter does all that? I am not quite sure u understand

**Reply by Fougl** [reply]
Do you maybe have info on what was written above?

**Reply by Fougl** [reply]
Then im guessing your answer is no, I cannot access the MISC folder

**Comment by Fougl** [comment]
I can confirm that this cable https://www.amazon.com/dp/B07XB4K4XC?psc=1&ref=ppx_yo2ov_dt_b_product_details does not allow you to add masks to MISC folder. MISC folder is only available when inserting solely SD card into a device. When connecting over USB it goes as usual into USB mode.

Here I found another issue with this approach. When using the following command !MTUSB=1!MBOOT="!Ltest"!SAVEtest=!SM3D4!R
If there will be a false positive during that time when camera is in USB mode, and you are downloading files, the motion detection mode will turn off. @dnewman-gpsw is there anything to do about it? @The-Skunk are you familiar with this problem?
